src/binary-reader-ir.cc

```diff
@@ -435,12 +422,10 @@ Result BinaryReaderIR::OnMemoryCount(Index count) {
 }
 
 Result BinaryReaderIR::OnMemory(Index index, const Limits* page_limits) {
-  ModuleField* field = module->AppendField();
-  field->loc = GetLocation();
-  field->type = ModuleFieldType::Memory;
-  field->memory = new Memory();
-  field->memory->page_limits = *page_limits;
-  module->memories.push_back(field->memory);
+  auto memory = new Memory();
+  memory->page_limits = *page_limits;
```

Comment:
Totally agree, at some point I'm picturing that a lot of this code will just be one-liners constructing everything. Kinda want to just fix it all now, but I'm trying to control my refactoring urges :-}